framework/src/media/audio/audio_manager.c

```diff
@@ -123,14 +131,12 @@ struct audio_samprate_map_entry_s {
 typedef enum audio_card_status_e audio_card_status_t;
 typedef enum manufacturer_type_e manufacturer_t;
 typedef enum audio_card_type_e audio_card_type_t;
+typedef enum audio_card_operation_e audio_card_operation_t;
 typedef struct audio_config_s audio_config_t;
 typedef struct audio_card_info_s audio_card_info_t;
 
-static audio_card_info_t g_audio_in_cards[CONFIG_AUDIO_MAX_INPUT_CARD_NUM];
-static audio_card_info_t g_audio_out_cards[CONFIG_AUDIO_MAX_OUTPUT_CARD_NUM];
-
-static int g_actual_audio_in_card_id = INVALID_ID;
-static int g_actual_audio_out_card_id = INVALID_ID;
+static int g_num_found_in_cards = 0;
```

Comment:
Why do we need this value?

Reply:
This is used in line 186 to check whether it is the first time to make a handler list.
Also in line 352, it is used to return an error if no cards are found.

Reply:
static variables are a problem.
init_audio_stream function can be checked